core-sdk-samples/higgs-shop-sample-app/src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable no-console */
+import { Dispatch, SetStateAction, useState } from 'react';
+
+type SetValue<T> = Dispatch<SetStateAction<T>>;
+
+function useLocalStorage<T>(
+    key: string,
+    initialValue: T,
+): [T, SetValue<T>, () => void] {
+    const [storedValue, setStoredValue] = useState<T>(() => {
+        if (typeof window === 'undefined') {
+            return initialValue;
+        }
```

Comment:
This is a defensive programming pattern I've seen in React whenever window is used in a component to prevent things breaking whenever using server side rendering. Since this is a custom hook that is intended to be reusable, I'm erring on being extra defensive, even if we're not using server side rendering now.